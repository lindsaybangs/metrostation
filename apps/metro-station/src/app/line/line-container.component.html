<div class="container">
<div *ngFor="let line of (this.statuses$ | async)">
    <lindsay-line [status]="line"></lindsay-line>
</div>
</div>