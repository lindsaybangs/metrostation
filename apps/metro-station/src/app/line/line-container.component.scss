div {
  display: flex;
  flex-flow: row wrap;
  align-content: center;
}
