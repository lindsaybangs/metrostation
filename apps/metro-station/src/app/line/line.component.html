<div>
    <h1>{{ this.status.line }}</h1>
    <p>{{ this.status.status }}</p>
</div>