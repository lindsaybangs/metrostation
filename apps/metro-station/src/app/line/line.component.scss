div {
  background: white;
  color: darkslategrey;
  min-width: 400px;
  min-height: 50px;

  margin: 10px;
  border: 1px solid darkslategrey;
  border-radius: 5px;
  box-shadow: 10px 10px 10px grey;
}

h1 {
  padding: 10px 25px;
  opacity: 1;
}

p {
  padding: 10px 25px;
  opacity: 1;
}
