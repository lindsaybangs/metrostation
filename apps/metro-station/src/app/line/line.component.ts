import { Component, Input, OnInit } from '@angular/core';

export interface MetroLine {
  line: string;
  status: string;
  start?: Date;
}

@Component({
  selector: 'lindsay-line',
  templateUrl: './line.component.html',
  styleUrls: ['./line.component.scss'],
})
export class LineComponent implements OnInit {
  @Input() status: MetroLine;

  constructor() {}

  ngOnInit(): void {}
}
