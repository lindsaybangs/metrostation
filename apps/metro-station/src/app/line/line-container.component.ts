import { Component, OnInit } from '@angular/core';
import { Observable, of } from 'rxjs';
import { MetroLine } from './line.component';
import { StmMetro, StmService } from '../stm.service';
import { map } from 'rxjs/operators';

@Component({
  selector: 'lindsay-line-container',
  templateUrl: './line-container.component.html',
  styleUrls: ['./line-container.component.scss']

})
export class LineContainerComponent implements OnInit {

    statuses$: Observable<MetroLine[]>;

    constructor(private metroService: StmService) { }
  
    ngOnInit(): void {
      this.getStatuses();
    }
  
    getStatuses(): void {
      this.statuses$ = this.metroService.fetch().pipe(map(this.mapMetroToLines));
    }
  
    mapMetroToLines(metro: StmMetro[]): MetroLine[] {
      return metro.map(line => { return { line: line.name, status: line.data.text }});
    }
}
