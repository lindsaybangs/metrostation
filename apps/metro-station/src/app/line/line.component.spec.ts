import { ComponentFixture, TestBed } from '@angular/core/testing';
import { LineComponent } from './line.component';

describe('LineComponent', () => {
  let fixture: ComponentFixture<LineComponent>;
  let component: LineComponent;

  beforeEach(() => {
    TestBed.configureTestingModule({
      providers: [LineComponent],
      declarations: [LineComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(LineComponent);
    component = TestBed.inject(LineComponent);
    component.status = { line: 'Orange', status: 'ok' };
  });

  it('should render with line status available', () => {
    expect(fixture).toMatchSnapshot();
  });
});
