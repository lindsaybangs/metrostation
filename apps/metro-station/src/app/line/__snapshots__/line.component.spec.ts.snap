// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineComponent should render with line status available 1`] = `
<lindsay-line>
  <div>
    <h1>
      
    </h1>
    <p>
      
    </p>
  </div>
</lindsay-line>
`;
